externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/connection/ConnectionProvider.scala

```diff
@@ -57,10 +57,9 @@ abstract class AbstractConnectionProvider extends Logging {
       case None =>
         // TODO
         if (filteredProviders.size != 1) {
-          throw new IllegalArgumentException(
-            "JDBC connection initiated but more than one connection provider was found. Use " +
-              s"${ENGINE_JDBC_CONNECTION_PROVIDER.key} option to select a specific provider. " +
-              s"Found active providers ${filteredProviders.mkString("[", ", ", "]")}")
+          warn("JDBC connection initiated but more than one connection provider was found. Use " +
```

Comment:
@bowenliang123 @pan3793 Both `MySQLConnectionProvider` and `DorisConnectionProvider` utilize `com.mysql.cj.jdbc.Driver`. This causes an exception. I have already changed it to a warn log here, will there be any problems?

Reply:
As both Doris and Mysql server are compatible the mysql driver, how about simply reuse `Mysql8ConnectionProvider` instead of introducing the new subclass `MySQLConnectionProvider` ?

Reply:
I believe using a distinct subclass of `MySQLConnectionProvider` would be better, as it helps users distinguish between different engines(MySQL,Doris). The presence of multiple connection providers does not impact user's experience, and shouldn't throw an exception and terminate the process. Instead, it should just remind users to set the `ENGINE_JDBC_CONNECTION_PROVIDER` parameter.

Reply:
> As both Doris and Mysql server are compatible the mysql driver, how about simply reuse `Mysql8ConnectionProvider` instead of introducing the new subclass `MySQLConnectionProvider` ?

This method seems good, but it may lead to confusion when users set the `kyuubi.engine.jdbc.connection.provider` parameter (since Doris and MySQL use the same provider). In contrast, I think that since the driver used by the provider is the same, it does not affect the user's usage. Users just need to be reminded to set the `kyuubi.engine.jdbc.connection.provider` parameter to specify the provider.

@pan3793 What do you think?

Reply:
@Kwafoor your suggestion LGTM, but I get no idea after reading the description of `kyuubi.engine.jdbc.connection.provider`, could you please enrich the docs?

```
The connection provider is used for getting a connection from the server
```

Reply:
How about that？

Reply:
the docs of `kyuubi.session.group.provider` should be a good example

Reply:
BTW, the `settings.md` is generated by `dev/gen/gen_all_config_docs.sh`

Reply:
> the docs of `kyuubi.session.group.provider` should be a good example

done